cmd/soroban-cli/src/commands/contract/init.rs

```diff
@@ -1,85 +1,76 @@
-use clap::{
-    builder::{PossibleValue, PossibleValuesParser, ValueParser},
-    Parser,
-};
-use gix::{clone, create, open, progress, remote};
-use rust_embed::RustEmbed;
-use serde_json::{from_str, json, to_string_pretty, Error as JsonError, Value as JsonValue};
 use std::{
-    env,
-    ffi::OsStr,
-    fs::{
-        copy, create_dir_all, metadata, read_dir, read_to_string, write, File, Metadata,
-        OpenOptions,
-    },
-    io::{self, Read, Write},
-    num::NonZeroU32,
+    fs::{create_dir_all, write},
+    io,
     path::{Path, PathBuf},
     str,
-    sync::atomic::AtomicBool,
 };
-use toml_edit::{Document, TomlError};
-use ureq::get;
 
-use crate::{commands::global, print};
+use clap::Parser;
+use rust_embed::RustEmbed;
 
-const SOROBAN_EXAMPLES_URL: &str = "https://github.com/stellar/soroban-examples.git";
-const GITHUB_URL: &str = "https://github.com";
-const WITH_EXAMPLE_LONG_HELP_TEXT: &str =
-    "An optional flag to specify Soroban example contracts to include. A hello-world contract will be included by default.";
+use crate::commands::contract::init::Error::{
+    AlreadyExists, PathExistsNotCargoProject, PathExistsNotDir,
+};
+use crate::{commands::global, print};
 
 #[derive(Parser, Debug, Clone)]
 #[group(skip)]
 pub struct Cmd {
     pub project_path: String,
```

Comment:
I think this needs to be updated to `Option<project_path>`. This will require either `project_path` or `--name` to be provided; if you run the command like `stellar contract init`, it should error.

Reply:
Isn't project path always required? Name is just specifying the name of the contract in the workspace, but presumably we still need to tell the tool where the workspace is.